src/pages/LoginPage.js

```diff
@@ -1,16 +1,102 @@
-import React from "react";
-import { Link } from "react-router-dom";
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+import "../styles/LoginPage.css";
 
-function LoginPage() {
+export default function Login() {
+  const [formData, setFormData] = React.useState({ email: "", password: "" })
+  const [item, setItems] = React.useState([])
+  const [formErrors, setFormErrors] = React.useState({});
+  const history = useNavigate();
+
+
+  React.useEffect(function () {
+    fetch('https://jsonplaceholder.typicode.com/users')
+      .then((response) => response.json())
+      .then(data => setItems(data))
+  }, [])
+
+
+  function handleChange(event) {
+    setFormData(prevFormData => {
+      return {
+        ...prevFormData,
+        [event.target.name]: event.target.value
+      };
+    });
+  }
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setFormErrors(validate(formData));
+  };
+
+  const validate = (values) => {
+    const errors = {};
+    const userid = formData.email;
+    const passid = formData.password;
+
+
+
+    item.map((items) => {
+      var paswd = items.username;
+      var num = 123;
+      paswd += num;
+      if (items.email !== userid && paswd !== passid) {
+        errors.username = "Invalid username or password";
+      }
+      else if (items.email === userid && paswd !== passid) {
+        errors.username = "Invalid username or passsword";
+      }
+      else if (items.email !== userid && paswd === passid) {
+        errors.username = "Invalid username or passsword";
+      }
+      else {
+        history("postDetails");
+      }
+    });
+    return errors;
+  };
   return (
-    <div>
-      <h1>This is Login Page</h1>
-      <Link to="postPage">Click to view Post page.</Link>
-      <br />
-      <Link to="postDetails">Click here to view Post Details.</Link>
-      <br />
-      <Link to="profilePage">Click here to view Profile Page.</Link>
+    <div className="container">
+      <h1>META</h1>
+
+      <form onSubmit={handleSubmit} classname="form-input">
+        <div></div>
+        <p>{formErrors.username}</p>
+        <label>
+          Username:
+
+          <input
+
+            className='user-input'
+            type="email"
+            name="email"
+            placeholder="Your email id"
+            onChange={handleChange}
+            value={formData.email}
+          />
+        </label>
+        <br></br>
```

Comment:
There is `fieldset` element in HTML where you can combine labels, inputs and errors. wrap these in it.

also user CSS instead or `<br />` tags

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset

Reply:
Used <br /> tags